### DISCUSSION TITLE
Please help with `colour.sd_gaussian` definition output.

### DISCUSSION THREAD

**Discussion by EricFlyMeToTheMoon** [colour-science/colour#1171] [Q&A] [answered]
I'm using "colour.sd_gaussian" for some project. I found it is not generating correct Gaussian distribution.

Here is my test code:
```
import colour
peak_wl = 555
fwhm = 20
sd1 = colour.sd_gaussian(peak_wl, fwhm, method="FWHM")
print(sd1[peak_wl])
print(sd1[peak_wl - fwhm/2])

1.0
0.778800783071
```

Based on how FWHM is defined, I think it should be 0.5 instead of 0.7788. Please help.

**Comment by thomasmansencal** [comment]
Hi @EricFlyMeToTheMoon,

I'm abroad with limited time to look at things, we follow and implement Yoshi Ohno formula, ~he refers to FWHM as *half spectral width*~:

![image](https://github.com/colour-science/colour/assets/99779/30709562-ba10-4ca2-9220-1b3ed2c00845)

Cheers,

Thomas

**Reply by EricFlyMeToTheMoon** [reply]
Hi Thomas,

My understanding is: Yoshi Ohno formula is a special fitting function in order to better match typical LED emission spectrum (typical Gaussian distribution doesn't match typical LED emission very well). Yoshi Ohno formula is quite different from Gaussian.

However, if "colour.sd_gaussian" is defined as "Returns: Gaussian spectral distribution.", it should use the Gaussian distribution instead of Yoshi Ohno formula, right?

Thanks for the kind reply.

Best,
Eric

**Reply by tjdcs** [reply]
I agree with Eric here. That paper excerpt refers to the "half width" whereas our api uses "full width half max". The ohno excerpt could be understood to be roughly the "half width half max". 

Solutions could be to rename our api to half width
add both `fwhm` and `hwhm`, and defer to `hwhm` by default / argument order. That way non-specific user code will remain the same, but using `fwhm` `kwarg` will be mathematically correct.

**Reply by thomasmansencal** [reply]
Ah yes you are both correct, some other options:

- Rename `sd_gaussian_fwhm` to `sd_half_spectral_width` and the argument to `hwhm`, this would NOT break the API.
- Keep `sd_gaussian_fwhm` but fix it to actually use `fwhm` and add a new `sd_half_spectral_width` definition, this would break existing behaviour but it seems like we have an issue here so a fix is in order.

**Comment by tjdcs** [comment]
Also if we don't have that LED model in the code, it could be very helpful.
- Tucker

**Reply by EricFlyMeToTheMoon** [reply]
Just some update here.

I tested "colour.sd_gaussian" with sigma (sigma = fwhm / 2.35482) as the parameter instead of FWHM; it seems generating correct Gaussian distribution in this way.

But if using FWHM as the parameter, it is not generating correct Gaussian distribution.

Anyway, I'm a new user here and I think this is a great python package. And thanks all for building it.

Regards,
Eric

```
import colour

peak_wl = 555
fwhm = 20

sd1 = colour.sd_gaussian(peak_wl, fwhm, method="FWHM")
print(sd1[peak_wl])
print(sd1[peak_wl - fwhm/2])

sigma = fwhm / 2.35482

sd2 = colour.sd_gaussian(peak_wl, sigma, method="Normal")
print(sd2[peak_wl])
print(sd2[peak_wl - fwhm/2])

1.0
0.778800783071
1.0
0.500000013255
```

**Comment by EricFlyMeToTheMoon** [comment]
Guys, I think this is a bug, after reading the source code.

The function sd_gaussian_fwhm is using a wrong formula of generating Gaussian distribution.

Gaussian distribution (https://mathworld.wolfram.com/GaussianFunction.html) should not directly use FWHM as input, but need to convert FWHM to Sigma first.

![image](https://github.com/colour-science/colour/assets/137454938/eb4d0f98-3fff-4166-83f8-464ae7a6d42a)


![image](https://github.com/colour-science/colour/assets/137454938/9cc252af-a28f-4e3d-bcc5-c414bba9f93a)

**Reply by thomasmansencal** [reply, marked as answer]
Fixed with https://github.com/colour-science/colour/pull/1184, merging soon!

**Reply by EricFlyMeToTheMoon** [reply]
Thank you and greatly appreciated!

**Reply by thomasmansencal** [reply]
My pleasure, sorry for the wait!
